### PR TITLE
fix: discounting not applied on sales invoice created by subscriptions

### PR DESCRIPTION
**Pull request by 0Pranav**
- fix option value in `Subscriptions`
- fix fetching `Apply Discount Based On` in Sales Invoice created by system using Subscriptions.

### CODE REVIEW

File: erpnext/accounts/doctype/subscription/subscription.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def create_invoice(self, prorate):
 
 		if self.additional_discount_percentage or self.additional_discount_amount:
 			discount_on = self.apply_additional_discount
-			invoice.apply_additional_discount = discount_on if discount_on else 'Grand Total'
+			invoice.apply_discount_on = discount_on if discount_on else 'Grand Total'
```
Comment (nabinhait) [inline erpnext/accounts/doctype/subscription/subscription.py:283]: Why this change?

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/subscription/subscription.py (hunk 1 above)
Comment (0Pranav) [reply erpnext/accounts/doctype/subscription/subscription.py:283]: there's no field called `apply_additional_discount` in Sales Invoice. The concerned field is named `apply_discount_on`. 

Screenshot 2020-01-27 at 3 48 33 PM

**Review by nabinhait — APPROVED** [review]
